File: 1sem/lab6/lab6/lab6.cpp
#include <iostream>
using namespace std;

int factorial(int n) {
    int factor = 1;
    for (int k = 1;k <= n; k++) {
        factor = factor * k;
    }
    return factor;
}


float ex(int x, float eps) {
    float e_add = 1, e_x = 1;
    for (int i = 1;e_add >= eps; i++) {
        e_add = pow(-1, i) * pow(x, i) / factorial(i);
        e_x += e_add;
    }
    return e_x;
}


int main()
{
    int x;
    float eps, y;
    cout << "Input x: ";
    cin >> x;
    cout << "Input eps: ";
    cin >> eps;
    if (x >= 0 || x <= 2) {
        y = ex(x, eps) + ex(2 * x, eps);
    }
    else if (x > 2) {
        y = (1 / ex(x + 5, eps)) - ex(x + 2, eps);
    }
    cout << "y = " << y;
    return 0;
}

File: 1sem/lab6/lab6.py
import math


def factorial(n):
    factor = 1
    for k in range(1, n + 1):
        factor = factor * k
    return factor


def ex(n):
    k = 1
    e_add = 1
    e_x = 1
    while math.fabs(e_add) >= eps:
        e_add = (-1 ** k) * ((n ** k) / factorial(k))
        e_x += e_add
        k += 1
    return e_x


x = int(input("Type x: "))
eps = float(input("Type eps: "))
if 0 <= x <= 2:
    y = ex(x) + ex(2 * x)
elif x > 2:
    y = (1 / ex(x + 5)) - ex(x + 2)
print("y =", y)
